cupy/core/_routines_math.pyx

```diff
@@ -213,35 +363,122 @@ cdef ndarray scan(ndarray a, ndarray out=None):
 
     """
     if a._shape.size() != 1:
-        raise TypeError('Input array should be 1D array.')
+        a = a.ravel()
```

Comment:
Why did you fix so?

Reply:
The actual scan operation on a ndim array returns a 1d array when no axis is specified, so we can just ravel the array.

Reply:
I agree that since we ravel it in `scan_core` we can just omit this change!